### DISCUSSION TITLE
How to set default values when creating comments?

### DISCUSSION THREAD

**Discussion by xrxrjiang** [microsoft/vscode-discussions#450] [Extension Development QnA] [answered]
When I create a comment, I want to be able to have default values, but the api doesn't provide the corresponding fields.
![image](https://user-images.githubusercontent.com/23227390/218028496-08d177b6-48ef-46f4-bd39-1cb837fb7148.png)

<img width="1076" alt="image" src="https://user-images.githubusercontent.com/23227390/218028419-146d6506-f147-4ee8-a2b3-fa9e6add799a.png">

**Comment by alefragnani** [comment, marked as answer]
Hi @xrxrjiang ,

There is no specific API for that _(default value)_, but you can define the an array of comments on this function. You could simply pass your default value as a single `Comment` in the `comments` param. Something like:

```ts

class NoteComment implements vscode.Comment {
	id: number;
	label: string | undefined;
	savedBody: string | vscode.MarkdownString; // for the Cancel button
	constructor(
		public body: string | vscode.MarkdownString,
		public mode: vscode.CommentMode,
		public author: vscode.CommentAuthorInformation,
		public parent?: vscode.CommentThread,
		public contextValue?: string
	) {
		this.id = ++commentId;
		this.savedBody = this.body;
	}
}
...
        const someComment = new NoteComment('body', vscode.CommentMode.Editing, { name: 'vscode' });
	commentController.createCommentThread(vscode.window.activeTextEditor!.document.uri, new vscode.Range(0, 0, 10, 10), [someComment]);
```

BTW, the `NoteComment` above is declared in VS Code Extension Samples repo (https://github.com/microsoft/vscode-extension-samples/tree/main/comment-sample).

Hope this helps

**Reply by xrxrjiang** [reply]
Thank you for your answer.
